src/rpc/pd/timestamp.rs

```diff
@@ -3,226 +3,185 @@
 //! This module is the low-level mechanisms for getting timestamps from a PD
 //! cluster. It should be used via the `get_timestamp` API in `PdClient`.
 
-use std::{
-    sync::{Arc, RwLock},
-    thread::{self, JoinHandle},
+use super::Timestamp;
+use crate::{Error, Result};
+
+use futures::{
+    channel::{mpsc, oneshot},
+    compat::*,
+    executor::block_on,
+    join, pin_mut,
+    prelude::*,
+    task::{Context, Poll, Waker},
 };
+use grpcio::{ClientDuplexReceiver, ClientDuplexSender, WriteFlags};
+use kvproto::pdpb::{PdClient, *};
+use std::{cell::RefCell, collections::VecDeque, pin::Pin, rc::Rc, thread};
+
+const MAX_PENDING_COUNT: usize = 64;
+
+/// The timestamp oracle which provides monotonically increasing timestamps.
+#[derive(Clone)]
+pub struct Tso {
+    /// The transmitter of a bounded channel which transports the sender of an oneshot channel to
```

Comment:
```suggestion
    /// The transmitter of a bounded channel which transports the sender of a oneshot channel to
```

Nit. English is a terrible language, sorry :-(